services/registry/src/sync/sync-grpc-server.ts

```diff
@@ -1,96 +1,97 @@
 import * as grpc from "grpc";
-import { combineLatest } from "rxjs";
+import { combineLatest, Observable } from "rxjs";
 import {
-  ConfigurationMessage,
-  Schema,
-  SubscribeParams,
-  Upstream,
-  UpstreamAuthentication,
-  UpstreamAuthCredentials
+    ConfigurationMessage,
+    Schema,
+    SubscribeParams,
+    Upstream,
+    UpstreamAuthentication,
+    UpstreamAuthCredentials
 } from "../generated/agogos_pb";
 import { IRegistryServer, RegistryService } from "../generated/agogos_grpc_pb";
 
 import syncSchema$ from "./sync-schemas";
 import syncUpstreams$ from "./sync-upstreams";
-import syncAuthProvider$ from "./sync-upstream-auth-credentials";
+import syncUpstreamAuthCredentials$ from "./sync-upstream-auth-credentials";
+import { AgogosConfiguration } from "./object-types";
 
 const PORT = process.env.GRPC_PORT || 4001;
 
 // TODO: make this more general
-const syncConfiguration$ = combineLatest(
-  [syncSchema$, syncUpstreams$, syncAuthProvider$],
-  (schema, upstreams, upstreamAuthCredentials) => ({
-    schema,
-    upstreams,
-    upstreamAuthCredentials
-  })
+const syncConfiguration$: Observable<AgogosConfiguration> = combineLatest(
+    [syncSchema$, syncUpstreams$, syncUpstreamAuthCredentials$],
+    (schema, upstreams, upstreamAuthCredentials) => ({
+        schema,
+        upstreams,
+        upstreamAuthCredentials
+    })
 );
 
 class GqlConfigurationSubscriptionServer implements IRegistryServer {
-  subscribe(call: grpc.ServerWriteableStream<SubscribeParams>) {
-    const subscription = syncConfiguration$.subscribe(
-      configuration => {
-        const gqlSchema = new Schema();
-        gqlSchema.setDefinition(configuration.schema);
+    subscribe(call: grpc.ServerWriteableStream<SubscribeParams>) {
+        const subscription = syncConfiguration$.subscribe(
+            (configuration: AgogosConfiguration) => {
+                const gqlSchema = new Schema();
+                gqlSchema.setDefinition(configuration.schema);
 
-        // TODO: extract to function
-        const upstreams: Upstream[] = Object.values(
-          configuration.upstreams
-        ).map(ep => {
-          const upstream = new Upstream();
-          upstream.setHost(ep.host);
+                // TODO: extract to function
+                const upstreams: Upstream[] = Object.values(
+                    configuration.upstreams
+                ).map(ep => {
+                    const upstream = new Upstream();
+                    upstream.setHost(ep.host);
 
-          const upstreamAuth = new UpstreamAuthentication();
-          // FIXME: get from ep
-          upstreamAuth.setAuthType(ep.auth.authType);
```

Comment:
Types say `auth`, CRD says `authType`
Verify this is correct please :)

Reply:
It works so I think all things are correct